### ISSUE TITLE
Getting "invalid format" when trying to add newly-generated keys

### ISSUE THREAD

**Issue by kwburnett** [PowerShell/Win32-OpenSSH#2056]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the latest version
- [X] Search the existing issues.

### Steps to reproduce

I am experiencing a problem with trying to add a newly-generated key. I'm testing on a new-ish install of Windows following [these steps](https://learn.microsoft.com/en-us/windows-server/administration/openssh/openssh_keymanagement). The SSH version I'm using was installed with `winget`.

I thought the issue might be related to [issue 612](https://github.com/PowerShell/openssh-portable/pull/612), but, unless it wasn't deployed with 9.2.2.0, it didn't seem to resolve my issue.

Here are the steps I performed on a test key:
```powershell
PS> # Set the sshd service to be started automatically
PS> Get-Service -Name sshd | Set-Service -StartupType Automatic
PS>
PS> # Now start the sshd service
PS> Start-Service sshd
PS> ssh-keygen -t ed25519
Generating public/private ed25519 key pair.
Enter file in which to save the key (C:\Users\username/.ssh/id_ed25519):
Enter passphrase (empty for no passphrase):
Enter same passphrase again:
Your identification has been saved in C:\Users\username/.ssh/id_ed25519
Your public key has been saved in C:\Users\username/.ssh/id_ed25519.pub
The key fingerprint is:
SHA256:NktHYgL/VHv0K7qMvWFVUljh6zOVkyyDDR61BqOdRW8 username@computer
The key's randomart image is:
+--[ED25519 256]--+
|    .     .oo*+. |
|     o   .ooB+o  |
|      o +.o=oo+E |
|       = o..*oooo|
|        S .oo+o=.|
|       o + o oo..|
|        . +   +  |
|         = o   o |
|        . =.     |
+----[SHA256]-----+
PS> # By default the ssh-agent service is disabled. Configure it to start automatically.
PS> # Make sure you're running as an Administrator.
PS> Get-Service ssh-agent | Set-Service -StartupType Automatic
PS>
PS> # Start the service
PS> Start-Service ssh-agent
PS>
PS> # This should return a status of Running
PS> Get-Service ssh-agent

Status   Name               DisplayName
------   ----               -----------
Running  ssh-agent          OpenSSH Authentication Agent

PS>
PS> # Now load your key files into ssh-agent
PS> ssh-add $env:USERPROFILE\.ssh\id_ed25519
Enter passphrase for C:\Users\username\.ssh\id_ed25519:
Could not add identity "C:\Users\username\.ssh\id_ed25519": invalid format
```

Output of `ssh -V`:
```powershell
PS> ssh -V
OpenSSH_for_Windows_9.2p1, LibreSSL 3.7.2
```

### Expected behavior

```console
PS> ssh-add $env:USERPROFILE\.ssh\id_ed25519
Enter passphrase for C:\Users\username\.ssh\id_ed25519:

__successful addition of SSH key___
```

### Actual behavior

```console
PS> ssh-add $env:USERPROFILE\.ssh\id_ed25519
Enter passphrase for C:\Users\username\.ssh\id_ed25519:
Could not add identity "C:\Users\username\.ssh\id_ed25519": invalid format
```

### Error details

```console
No output from `Get-Error`.
```

### Environment data

```PowerShell
Name                           Value
----                           -----
PSVersion                      7.3.3
PSEdition                      Core
GitCommitId                    7.3.3
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Version

9.2.2.0 (though I had the same problem on 8.6.0.0 and 8.9.1.0)

**Comment by tgauth** [comment]
I'm not sure this would have an impact, but any chance there's some sort of version mismatch between ssh-keygen & ssh-agent? 

Could you elaborate on the potential connection to https://github.com/PowerShell/openssh-portable/pull/612? Seeing as that was a hardware-backed key, it's not immediately clear to me. 
Is ed25519 the only key type that isn't working, or do others (rsa, ecdsa) have the same behavior?

Also, debug logs from the ssh-agent would be helpful.
It can be run in debug mode similar to the steps described here for sshd: https://github.com/PowerShell/Win32-OpenSSH/wiki/Troubleshooting-Steps

**Comment by kwburnett** [comment]
Today it started working! I tried going to add an rsa key and kept getting the `Error connecting to agent: No such file or directory` message. Both `sshd` and `ssh-agent` were running, and stopping/restarting them did nothing.

Then I looked at my user (not System) environment variables and saw that both `SSH_AUTH_SOCK` and `SSH_AGENT_PID` were defined (probably from a time in the past when I ran SSH differently). Clearing those two environment variables and restarting my system solved the problem. Thank you very much!

Just to answer your question about the link to PR 612, it was a very loose connection in that I saw "invalid format" was being generated there, too. I didn't spend too much time on the particulars...sorry about that.
